include/cppkafka/message_builder.h

```diff
@@ -234,23 +243,50 @@ template <typename T, typename C>
 BasicMessageBuilder<T, C>::BasicMessageBuilder(const Message& message)
 : topic_(message.get_topic()),
   key_(Buffer(message.get_key().get_data(), message.get_key().get_size())),
+#if (RD_KAFKA_VERSION >= RD_KAFKA_HEADERS_SUPPORT_VERSION)
+  header_list_(message.get_header_list() ?
+               HeaderListType(rd_kafka_headers_copy(message.get_header_list().get_handle())) : HeaderListType()), //copy headers
```

Comment:
Similar to the comment about moving, `HeaderList` is copyable, right? Does this achieve something different than what `header_list_(message.get_header_list())` does? Maybe I'm missing something.

Reply:
In this function I can't just call the move or the copy constructor because you're copying/moving from `HeaderList<Header<T>>` to `HeaderList<Header<U>>` and there's no conversion constructor so you have to pass via the raw handle. Otherwise I can implement a copy constructor from `HeaderList<A>` to `HeaderList<B>` in which case you can then use the move or copy directly. Let me know if you prefer that...